cpp/src/arrow/compute/row/compare_test.cc

```diff
@@ -164,5 +166,128 @@ TEST(KeyCompare, CompareColumnsToRowsTempStackUsage) {
   }
 }
 
+// Compare columns to rows at offsets over 2GB within a row table.
+// Certain AVX2 instructions may behave unexpectedly causing troubles like GH-41813.
+TEST(KeyCompare, CompareColumnsToRowsLarge) {
```

Comment:
What is the runtime of this test? Perhaps we need to disable it on Valgrind builds.

Reply:
What do you mean by "runtime"? I can't think of a reason why Valgrind would complain (at least ASAN didn't).

Reply:
Sorry, I meant "run time" or execution time :-)

Reply:
Ah, got it! It takes about 20s with ASAN enabled. Perhaps it will be fine with Valgrind too?

Reply:
I should take a quick look.

Reply:
Ok, it takes 70s locally under Valgrind. That's a bit high for a single test, I would rather disable it under Valgrind.

Reply:
OK. Updated to disable the test under Valgrind. Thanks for helping running in your local!